pennylane/tape/tapes/tape.py

```diff
@@ -704,7 +704,7 @@ def set_parameters(self, params, trainable_only=True):
         [4, 1, 6]
         """
         if trainable_only:
-            iterator = zip(self.trainable_params, params)
+            iterator = zip(sorted(self.trainable_params), params)
```

Comment:
How come that sortedness only mattered here, wouldn't we want the sorting to happen always when querying the trainable parameters? So as to change the def of the `trainable_params` property to `return sorted(self._trainable_params)`, instead of sorting only here.

Reply:
Hey @antalszava, in fact, this is the only place in the code base where the order of `self.trainable_params` matters. In other parts of the code base we _do_ iterate over `self.trainable_params`, but we do not zip it with ordered data like we do here.

I suppose this was just me being cautious wrt overhead - rather than sorting + converting to a list everywhere `self.trainable_params` is called, I'm only doing it here to avoid additional overhead.

Reply:
That's good to know, thanks! Yes, I also think that the overhead aspect necessitates that sorting only happens at certain times.

One thing that could be nice is to leave a comment explaining why the sorting is needed here, but not elsewhere.